### PR TITLE
feat: `styleIncludePaths` for feature parity with Angular CLI

### LINKED ISSUE #282: Feature parity to angular-cli: `stylePreprocessorOptions.includePaths`
## Type of Issue

```
[ ] Bug Report
[x ] Feature Request
```

## Description

First let me say a big thank you for your work!

Angular-cli allows to add include-paths for the style-preprocessor in use:

```
      "stylePreprocessorOptions": {
        "includePaths": [
          "styles"
        ]
      },
```

In the scss / less files, you don't have to provide the complete relative path to an import.

#### How To Reproduce

Add the includePath to angular-cli.json and add an import to a file in your styles-folder without specifying the complete relative path.

#### Expected Behaviour

As angular-cli itself resolves the import correctly, ng-packagr should do it, too.

#### Version Information

```
ng-packagr: v1.6.0
node: v8.7.0
@angular: v5.0.0
```

### PR DESCRIPTION
**Pull request by mwootendev**
## I'm submitting a...

```
[ ] Bug Fix
[X] Feature
[ ] Other (Refactoring, Added tests, Documentation, ...)
```

## Checklist

- [X] Commit Messages follow the [Conventional Commits](https://conventionalcommits.org/) pattern
  - A feature commit message is prefixed "feat:"
  - A bugfix commit message is prefixed "fix:"
- [X] Tests for the changes have been added

## Description

Updating the ng-package.json file to include a more generic option that also supports resolving Less and Stylus import paths to achieve feature parity with `@angular/cli`.

- Added styleIncludePaths option in ng-package.schema.json
- Updated README.md to refer to the new option name
- Updated entry-point.ts to use the new option (also uses older sassIncludePaths for backwards compatability).
- Updated stylesheet.transform.ts to pass the include paths to the Less and Stylus compilers.
- Updated the integration tests for scss-paths to also include tests for Less and Stylus

This should resolve the comments about feature parity in #282.

Closes #282 

## Does this PR introduce a breaking change?

```
[ ] Yes
[X] No
```

I left the lib.sassIncludePaths option in place for backwards compatability

### CODE REVIEW

File: integration/samples/scss-paths/baz/baz.component.ts [hunk 1]
```diff
@@ -3,7 +3,6 @@ import { Component } from '@angular/core';
 @Component({
   selector: 'baz-component',
   templateUrl: './baz.component.html',
-  styleUrls: ['./baz.component.scss']
+  styleUrls: ['./baz.component.scss', './baz.component.less', './baz.component.styl']
```
Comment (dherges) [inline integration/samples/scss-paths/baz/baz.component.ts]: Very good idea to put all three style preprocessors here!

File: integration/samples/scss-paths/sub-module/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "$schema": "../../../../src/package.schema.json",
-  "name": "@sample/secondary/sub-module",
+  "name": "@sample/scss-paths/sub-module",
```
Comment (dherges) [inline integration/samples/scss-paths/sub-module/package.json]: Good catch!

File: integration/samples/scss-paths/sub-module/bar/bar.component.styl [hunk 3]
```diff
@@ -0,0 +1,5 @@
+@import "theme"
+@require "common"
```
Comment (dherges) [inline integration/samples/scss-paths/sub-module/bar/bar.component.styl]: I never used stylus before. Is that common synatx with `@import` and `@require`? Where is the difference between the two?

File: src/lib/ng-package-format/entry-point.ts [hunk 4]
```diff
@@ -95,8 +95,9 @@ export class NgEntryPoint {
     return this.$get('lib.flatModuleFile') || this.flattenModuleId('-');
   }
 
-  public get sassIncludePaths(): string[] {
-    const includePaths = this.$get('lib.sassIncludePaths') || [];
+  public get styleIncludePaths(): string[] {
+    // lib.sassIncludePaths retained for backwards compatability
+    const includePaths = this.$get('lib.styleIncludePaths') || this.$get('lib.sassIncludePaths') || [];
```
Comment (dherges) [inline src/lib/ng-package-format/entry-point.ts]: 👍

File: integration/samples/scss-paths/sub-module/bar/bar.component.styl (hunk 3 above)
Comment (mwootendev) [reply integration/samples/scss-paths/sub-module/bar/bar.component.styl]: I've not used stylus before either. From what I read they are mostly equivalent for local project imports, so I included examples of both to make sure they both work.

Comment (dherges) [reply integration/samples/scss-paths/sub-module/bar/bar.component.styl]: FOr documentation on stylus @import and @require: http://stylus-lang.com/docs/import.html

**Pull request merged**